video_xblock/video_xblock.py

```diff
@@ -0,0 +1,230 @@
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode("utf8")
+
+    def match(self, href):
+        if isinstance(self.url_re, list):
+            return any(regex.search(href) for regex in self.url_re)
+        elif isinstance(self.url_re, re.RegexObject):
+            return self.url_re.search(href)
+        elif isinstance(self.url_re, basestring):
+            return re.search(self.url_re, href, re.I)
+
+
+class DummyPlayer(BaseVideoPlayer):
+    @property
+    def url_re(self):
+        return [re.compile(r'')]
+
+    def get_frag(self, **context):
+        return Fragment(u'[Here be Video]')
+
+    def media_id(self, href):
+        return '<media_id>'
+
+
+class YoutubePlayer(BaseVideoPlayer):
+    # http://regexr.com/3a2p0
+    url_re = re.compile(r'(?:youtube\.com\/\S*(?:(?:\/e(?:mbed))?\/|watch\?(?:\S*?&?v\=))|youtu\.be\/)(?P<media_id>[a-zA-Z0-9_-]{6,11})')
+
+    def media_id(self, href):
+        return self.url_re.search(href).group('media_id')
+
+    def get_frag(self, **context):
+        html = Template(self.resource_string("static/html/youtube.html"))
+        frag = Fragment(
+            html_parser.unescape(
+                html.render(Context(context))
+            )
+        )
+
+        frag.add_css(self.resource_string(
+            'static/bower_components/video.js/dist/video-js.min.css'
+        ))
+        frag.add_javascript(self.resource_string(
+            'static/bower_components/video.js/dist/video.min.js'
+        ))
+        frag.add_javascript(self.resource_string(
+            'static/bower_components/videojs-youtube/dist/Youtube.min.js'
+        ))
+
+        return frag
+
+
+class BrightcovePlayer(BaseVideoPlayer):
+    url_re = re.compile(r'https:\/\/studio.brightcove.com\/products\/videocloud\/media\/videos\/(?P<media_id>\d+)')
+
+    def media_id(self, href):
+        return self.url_re.match(href).group('media_id')
+
+    def get_frag(self, **context):
+        html = Template(self.resource_string("static/html/brightcove.html"))
+        frag = Fragment(
+            html_parser.unescape(
+                html.render(Context(context))
+            )
+        )
+        return frag
+
+
+class VideoXBlock(StudioEditableXBlockMixin, XBlock):
+
+    icon_class = "video"
+
+    display_name = String(
+        default='Video',
+        display_name=_('Component Display Name'),
+        help=_('The name students see. This name appears in the course ribbon and as a header for the video.'),
+        scope=Scope.content,
+    )
+
+    href = String(
+        default='',
+        display_name=_('Video URL'),
+        help=_('URL of the video page. E.g. https://example.wistia.com/medias/12345abcde'),
+        scope=Scope.content
+    )
+
+    account_id = String(
+        default='',
+        display_name=_('Account Id'),
+        help=_('Your Brightcove account id'),
+        scope=Scope.content,
+    )
+
+    player_name = String(
+        default='dummy-player',
```

Comment:
> For cases when we cannot display useful player. I.e. empty video URL.

Please leave a comment about this in the code. Docstring for the class, for example.